### PR TITLE
Implement edit user endpoint

### PR DESCRIPTION
**Pull request by coldmagpie**

### CODE REVIEW

File: Queueomatic/Server/Endpoints/User/Delete/DeleteUserEndpoint.cs [hunk 1]
```diff
@@ -15,11 +15,19 @@ public DeleteUserEndpoint(IUnitOfWork unitOfWork)
...
     {
         Delete("/users/{email}");
+        Roles("User");
         Policies("SignedInUser");
     }
 
     public override async Task HandleAsync(DeleteUserRequest req, CancellationToken ct)
     {
+
+        if (!User.IsInRole("User") && !req.Email.Equals(req.UserId))
```
Comment (FabianAronsson) [inline Queueomatic/Server/Endpoints/User/Delete/DeleteUserEndpoint.cs:25]: If only a user can access this endpoint, then why do you need to check if the user has that role?

That is already done in the policy.

File: Queueomatic/Server/Endpoints/User/Edit/EditUserEndpoint.cs [hunk 2]
```diff
@@ -1,32 +1,52 @@
...
+    {
+        _unitOfWork = unitOfWork;
+    }
     public override void Configure()
     {
-        Verbs(Http.PUT);
-        Routes("/users/{email}");
-        AllowAnonymous();
+        Put("/users/{email}/update/profile");
+        Roles("User");
```
Comment (FabianAronsson) [inline Queueomatic/Server/Endpoints/User/Edit/EditUserEndpoint.cs:18]: The policy already checks this.

File: Queueomatic/Server/Endpoints/User/Edit/EditUserEndpoint.cs [hunk 3]
```diff
@@ -1,32 +1,52 @@
...
+        Policies("SignedInUser");
     }
 
     public override async Task HandleAsync(EditUserRequest req, CancellationToken ct)
     {
-        try
+        if (!User.IsInRole("User") && !req.Email.Equals(req.UserId))
```
Comment (FabianAronsson) [inline Queueomatic/Server/Endpoints/User/Edit/EditUserEndpoint.cs:24]: Again, if the user is already authenticated in the policy you don't need to check if it has the role.

**Review by FabianAronsson — CHANGES_REQUESTED** [review]
Check my comments.

File: Queueomatic/Server/Endpoints/User/Edit/EditUserEndpoint.cs [hunk 4]
```diff
@@ -1,32 +1,52 @@
...
+        var result = await _unitOfWork.SaveAsync();
+        
+        if (result == 0)
         {
-            Logger.LogInformation($"Task {nameof(EditUserEndpoint)} was cancelled.");
+            await SendErrorsAsync();
+            return;
         }
+
+        await SendAsync(new(new UserDto
```
Comment (Mojtabarezaei4) [inline Queueomatic/Server/Endpoints/User/Edit/EditUserEndpoint.cs:46]: Checkout the API specification for the actual response code.

**Review by Mojtabarezaei4 — CHANGES_REQUESTED** [review]
Please check my comment too.

PS. Try to make a pull request for the files which are related to each other. It makes it much easier to understand the pull request.

File: Queueomatic/Server/Endpoints/User/Edit/EditUserEndpoint.cs (hunk 4 above)
Comment (coldmagpie) [reply Queueomatic/Server/Endpoints/User/Edit/EditUserEndpoint.cs:40]: what do you mean this?

Comment (Mojtabarezaei4) [reply Queueomatic/Server/Endpoints/User/Edit/EditUserEndpoint.cs:40]: This endpoint should respond with no content.

Comment (FabianAronsson) [reply Queueomatic/Server/Endpoints/User/Edit/EditUserEndpoint.cs:40]: > Checkout the API specification for the actual response code.

Is the api specification not incorrect here? We are specifying a 204 which means no content, but you are supposed to return a user as response body. Are those two not contradicting?

**Comment by FabianAronsson** [comment]
> Please check my comment too.
> 
> PS. Try to make a pull request for the changes/files which are related to each other. It makes it much easier to understand the pull request.

I allowed it this time. It was a small fix.

**Review by FabianAronsson — APPROVED** [review]

File: Queueomatic/Server/Endpoints/User/Edit/EditUserEndpoint.cs (hunk 4 above)
Comment (Mojtabarezaei4) [reply Queueomatic/Server/Endpoints/User/Edit/EditUserEndpoint.cs:40]: It can be both 200 and 204. Here makes changes and the response will not be used.

**Review by Mojtabarezaei4 — APPROVED** [review]

**Pull request merged**